contracts/bonding/BondingManager.sol

```diff
@@ -498,41 +529,133 @@ contract BondingManager is IBondingManager, Controllable {
+     */
+    function mintedTokensPerReward() public constant returns (uint256) {
+        return initialTokenSupply.mul(initialYearlyInflation).div(100).div(roundsManager().rewardCallsPerYear());
     }
 
     /*
-     * @dev Computes rewards for a delegator since last state transition
+     * @dev Computes token distribution for delegator since its last state transition
      * @param _delegator Address of delegator
      */
-    function delegatorRewards(address _delegator) public constant returns (uint256) {
-        uint256 rewards = 0;
+    function delegatorTokenPoolsShare(Delegator storage del) internal constant returns (uint256) {
+        uint256 tokens = 0;
 
-        // Check if delegator bonded to a transcoder
-        if (delegators[_delegator].transcoderAddress != address(0)) {
+        if (del.transcoderAddress != address(0)) {
             // Iterate from round that delegator last transitioned states to current round
-            // If the delegator is bonded to a transcoder, it has been bonded to the transcoder since lastStateTransitionRound
-            for (uint256 i = delegators[_delegator].lastStateTransitionRound; i <= roundsManager().currentRound(); i++) {
-                uint256[2] rewardMultiplier = rewardMultiplierPerTranscoderAndRound[delegators[_delegator].transcoderAddress][i];
+            // If the delegator is bonded to a transcoder, it has been bonded to the transcoder since lastStakeUpdateRound
+            for (uint256 i = del.lastStakeUpdateRound; i <= roundsManager().currentRound(); i++) {
+                tokens = tokens.add(delegatorRewardPoolShare(del, i)).add(delegatorFeePoolShare(del, i));
+            }
+        }
 
-                // Check if transcoder has a reward multiplier for this round (total minted tokens for round > 0)
-                if (rewardMultiplier[0] > 0) {
-                    // Calculate delegator's share of reward
-                    uint256 delegatorShare = rewardMultiplier[0].mul(delegators[_delegator].bondedAmount).div(rewardMultiplier[1]);
+        return tokens;
+    }
+
+    /*
+     * @dev Computes delegator's share of reward pool for a round
+     */
+    function delegatorRewardPoolShare(Delegator storage del, uint256 _round) internal constant returns (uint256) {
+        RewardPool storage rewardPool = transcoders[del.transcoderAddress].tokenPoolsPerRound[_round].rewardPool;
 
-                    rewards = rewards.add(delegatorShare);
+        if (rewardPool.rewards == 0) {
+            return 0;
+        } else {
+            return rewardPool.rewards.mul(del.bondedAmount).div(rewardPool.transcoderTotalStake);
+        }
+    }
+
+    /*
+     * @dev Computes delegator's share of fee pool for a round
+     */
+    function delegatorFeePoolShare(Delegator storage del, uint256 _round) internal constant returns (uint256) {
+        ClaimFees[] storage feePool = transcoders[del.transcoderAddress].tokenPoolsPerRound[_round].feePool;
```

Comment:
Let's do a quick in person walkthrough of the feePool and tokenPoolsPerRound structures.